src/tools/wasm-split/CMakeLists.txt

```diff
@@ -6,5 +6,5 @@ set(wasm_split_SOURCES
 )
 add_library(wasm-split-lib OBJECT ${wasm_split_SOURCES})
```

Comment:
Is there some reason we need `OBJECT` where?  I would have though `STATIC` makes more sense is the more common way to do things.

Reply:
I tried STATIC, but it failed on the mingw builder. I was probably just holding it wrong, but this approach worked. Here's the failing log: https://github.com/WebAssembly/binaryen/runs/3539498459.

Reply:
Ah, I'll bet that's a link order issue. When using GNU ld, each archive is checked only once, in the order they appear on the linker command line. If an object file is needed to satisfy an undefined reference, it gets added to the link. But if another object is added later and has an undefined reference to an object in the earlier archive (that wasn't previously added), then you can get this kind of error. lld and link.exe will basically iterate over all the libraries until convergence.

So to fix this, we could fix the link order of the libraries (or fix any legit circular dependencies), or we can just give ld the same behavior as the others by adding `--start-group` and `--end-group` around the relevant sets of libraries. I wonder if there's an easy way to make CMake do that...